library/core/src/lib.rs

```diff
@@ -154,6 +154,7 @@
 #![feature(cfg_target_has_atomic)]
 #![feature(cfg_target_has_atomic_equal_alignment)]
 #![feature(cfg_ub_checks)]
+#![feature(const_destruct)]
```

Comment:
@compiler-errors @lcnr @fee1-dead is this feature okay to use in libcore?

Reply:
Not until the syntax is finalized by T-lang. I've marked this PR as blocked

Reply:
Which syntax? There is no `~const` in this PR.

Reply:
All this feature gate does is
- allow `impl const Drop` -- but we already have a few `impl const` in the standard library so that syntax can't be the concern
- adjust the const_check logic to allow `drop` in const if the type in question is const-destructible

Reply:
I think we'll just hold off on merging. I'm optimistic we'll get a decision soon